userspace/libsinsp/container.cpp

```diff
@@ -330,7 +330,15 @@ void sinsp_container_manager::notify_new_container(const sinsp_container_info& c
 		//
 		// Fallback at just storing the new container.
 		// NOTE: this must be kept in sync with what happens on container event parsing, in parsers.cpp.
-		add_container(std::make_shared<sinsp_container_info>(container_info), tinfo);
+		const auto container = m_inspector->m_container_manager.get_container(container_info.m_id);
```

Comment:
Just a nitpick: what about adding this logic buried inside `add_container`?

Reply:
I'm not particularly fond of it, since e.g. when parsing container JSON events this check is quite far away from add_container (we construct the container object in the meantime, so running this check early saves us some work).

We can always duplicate it but I'm not sure it makes it clearer.

Reply:
I don't know, i find it better to have that kind of logic where it should belong, ie: while adding the element to the set.
But i am ok with this solution, no strong opinion on that.

Reply:
I agree in principle, but this case doesn't want to be too principled :( (we can either do it cleanly the way you're proposing, or we can avoid a bunch of JSON parsing and building objects in the hot path)

Reply:
+1 then :)